core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -54,12 +54,30 @@ import org.apache.spark.util.ArrayImplicits._
 class DAGSchedulerEventProcessLoopTester(dagScheduler: DAGScheduler)
   extends DAGSchedulerEventProcessLoop(dagScheduler) {
 
+  dagScheduler.setEventProcessLoop(this)
```

Comment:
To make sure I am not missing anything, this is to handle case of a `DAGSchedulerEvent` processing resulting in another `DAGSchedulerEvent` getting generated (as part of the logic for the event) - and we want the first to complete before the second is handled (which is how 'normal' flow would be).

Is this right ?

Reply:
Right. Before this change, the the event process loop are derived. There are two event process loops involved in the testing DAGScheduler (i.e., `MyDAGScheduler`). One is the manually created - `DAGSchedulerEventProcessLoopTester` and another one is the default `DAGSchedulerEventProcessLoop` that created inside `MyDAGScheduler`'s base class `DAGScheduler.` And any explicitly created events that posted during test using `DAGSchedulerEventProcessLoopTester`. And other implictly created events (inside `DAGScheduler`) are posted by `DAGSchedulerEventProcessLoop`. This leads to the events being processed by two separate threads so the "first to complete before the second" policy can no longer be guaranteed. This change ensures we only use one event process loop during the test.

Reply:
The example is the test `DAGSchedulerSuite - trivial job failure`. In the test, it explictly post `TaskSetFailed` by `DAGSchedulerEventProcessLoopTester`. While processing `TaskSetFailed`, it generates event `StageFailed` (added by this PR) that posted by `DAGSchedulerEventProcessLoop`.

Reply:
Sounds good, thanks for clarifying !